### ISSUE TITLE
False-positive W0236 (and W0221) when implementing an optional protocol method

### ISSUE THREAD

**Issue by bersbersbers** [pylint-dev/pylint#8687]
### Bug description

pylint does not accept a method for a property when inheriting from a protocol. Closely related: https://github.com/python/mypy/issues/15243

```python
"""Bug."""
from typing import Callable, Protocol

class Proto(Protocol):
    @property
    def fun(self) -> Callable[[int], None] | None:
        ...

class Class(Proto):
    def fun(self, i: int) -> None:
        print(i)
```

### Command used

```shell
pylint bug.py
```

### Pylint output

```shell
************* Module bug
bug.py:10:4: W0221: Number of parameters was 1 in 'Proto.fun' and is now 2 in overriding 'Class.fun' method (arguments-differ)
bug.py:10:4: W0236: Method 'fun' was expected to be 'property', found it instead as 'method' (invalid-overridden-method)

-------------------------------------------------------------------
Your code has been rated at 7.14/10 (previous run: 10.00/10, -2.86)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)]
```

### OS / Environment

Windows 10
